autonomy/cli/helpers/analyse.py

```diff
@@ -349,39 +356,41 @@ def _get_ipfs_pins(is_on_chain_check: bool = False) -> Set[str]:
     return set()
 
 
-def _get_service_hash(service_id: PackageId, package_manager: PackageManagerV1) -> str:
-    """Iterate through the available packages and find the hash for the service"""
-
-    for package_id, package_hash in package_manager.dev_packages.items():
-        if (
-            package_id.package_type == PackageType.SERVICE
-            and package_id.public_id.to_any() == service_id.public_id.to_any()
-        ):
-            return package_hash
-
-    raise click.ClickException(
-        f"Service with public id `{service_id.public_id}` not found"
-    )
-
-
-def check_service_readiness(
+def check_service_readiness(  # pylint: disable=too-many-locals
     token_id: Optional[int],
-    service_id: PackageId,
+    public_id: Optional[PublicId],
     chain_type: ChainType,
     packages_dir: Path,
 ) -> None:
     """Check deployment readiness of a service."""
 
-    package_manager = PackageManagerV1.from_dir(packages_dir=packages_dir)
     is_on_chain_check = token_id is not None
+    ledger_api, _ = get_ledger_and_crypto_objects(chain_type=chain_type)
+    package_manager = PackageManagerV1.from_dir(packages_dir=packages_dir)
     ipfs_pins = _get_ipfs_pins(is_on_chain_check=is_on_chain_check)
 
-    service_id = service_id.with_hash(
-        package_hash=_get_service_hash(
-            service_id=service_id,
-            package_manager=package_manager,
-        )
-    )
+    if is_on_chain_check:
+        try:
+            service_info = resolve_component_id(
+                ledger_api=ledger_api,
+                contract_address=ContractConfigs.service_registry.contracts[chain_type],
+                token_id=cast(int, token_id),
+                is_service=True,
+            )
+            package_hash = cast(str, service_info["code_uri"]).replace("ipfs://", "")
+            public_id = PublicId.from_str(service_info["name"]).with_hash(
+                package_hash=package_hash
+            )
+            service_id = PackageId(
+                package_type=PackageType.SERVICE, public_id=public_id
+            )
+        except FailedToRetrieveComponentMetadata as e:
+            raise click.ClickException(
+                f"On chain service verification failed; {e}"
+            ) from e
+    else:
+        service_id = PackageId(package_type=PackageType.SERVICE, public_id=public_id)
```

Comment:
When performing the on-chain check we need to use the on-chain hash, the implementation used the local hash before this